dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/loadbalance/AbstractLoadBalance.java

```diff
@@ -45,12 +44,6 @@ static int calculateWarmupWeight(int uptime, int warmup, int weight) {
 
     @Override
     public <T> Invoker<T> select(List<Invoker<T>> invokers, URL url, Invocation invocation) {
-        if (CollectionUtils.isEmpty(invokers)) {
-            return null;
-        }
-        if (invokers.size() == 1) {
-            return invokers.get(0);
-        }

```

Comment:
I have a puzzle about why you put these empty check codes into subclasses, Can you explain that?

Reply:
To make all the `select` logic extendable.